### PR TITLE
Add encoding/decoding for account module

### PR DESCRIPTION
**Pull request by willclarktech**
Part 1 of #1340

Note: no tests are currently included! Also I've implemented all the timestamps as vanilla `number`s. I believe they are Unix timestamps in seconds so this should be fine, but I have a general worry about the type conversion from `int64`.

### CODE REVIEW

File: packages/iov-bns/types/types.d.ts [hunk 1]
```diff
@@ -283,6 +315,101 @@ export interface TransferUsernameTx extends UnsignedTransaction {
...
+}
+export declare function isUpdateAccountConfigurationTx(
+  tx: UnsignedTransaction,
+): tx is UpdateAccountConfigurationTx;
+export interface RegisterDomainTx extends UnsignedTransaction {
+  readonly kind: "bns/register_domain";
+  readonly domain: string;
+  readonly admin: Address;
+  readonly hasSuperuser: boolean;
+  readonly thirdPartyToken?: Uint8Array;
```
Comment (davepuchyr) [inline packages/iov-bns/types/types.d.ts]: This should be an `Address`.

File: packages/iov-bns/src/encodemsg.spec.ts [hunk 2]
```diff
@@ -159,6 +174,208 @@ describe("encodeMsg", () => {
...
+
+  it("works for UpdateAccountConfigurationTx", () => {
+    const transfer: UpdateAccountConfigurationTx = {
+      kind: "bns/update_account_configuration",
+      chainId: defaultChainId,
+      configuration: {
+        owner: "tiov1p62uqw00znhr98gwp8vylyyul844aarjhe9duq" as Address,
+        validDomain: "hole",
+        validName: "rabbit",
+        validBlockchainId: "wonderland",
```
Comment (albertandrejev) [inline packages/iov-bns/src/encodemsg.spec.ts]: :+1:

File: scripts/bnsd/genesis_app_state.json [hunk 3]
```diff
@@ -165,6 +191,14 @@
       "owner": "cond:gov/rule/0000000000000002",
       "valid_username_name": "^[a-z0-9\\-_.]{3,64}$",
       "valid_username_label": "^iov$"
+    },
+    "account": {
+      "valid_name": "^[a-z0-9\\-_.]{3,64}$",
+      "valid_domain": "^[a-z0-9\\-_]+${3,16}$",
```
Comment (albertandrejev) [inline scripts/bnsd/genesis_app_state.json]: Please remove `+$` from regexp.
It should looks like `"^[a-z0-9\\-_]{3,16}$"`

File: scripts/bnsd/start.sh [hunk 4]
```diff
@@ -5,7 +5,7 @@ command -v shellcheck > /dev/null && shellcheck "$0"
 # Choose from https://hub.docker.com/r/iov1/tendermint/tags/
 export BNSD_TM_VERSION=v0.31.5
```
Comment (albertandrejev) [inline scripts/bnsd/start.sh]: Update version to `v0.31.11-iov1`

File: packages/iov-bns/src/bnsconnection.posttxs.spec.ts [hunk 5]
```diff
@@ -603,6 +612,275 @@ describe("BnsConnection (post txs)", () => {
       connection.disconnect();
     });
 
+    it("can register a domain", async () => {
+      pendingWithoutBnsd();
+      const connection = await BnsConnection.establish(bnsdTendermintUrl);
+      const registryChainId = connection.chainId();
```
Comment (webmaster128) [inline packages/iov-bns/src/bnsconnection.posttxs.spec.ts]: This chain ID getter recently changed from a function to a property (https://github.com/iov-one/iov-core/pull/1376). So it must be

```ts
const registryChainId = connection.chainId;
```

File: packages/iov-bns/src/bnsconnection.posttxs.spec.ts [hunk 6]
```diff
@@ -603,6 +612,275 @@ describe("BnsConnection (post txs)", () => {
...
+      const firstSearchResultTransaction = searchResult[0].transaction;
+      assert(isRegisterDomainTx(firstSearchResultTransaction), "Expected RegisterDomainTx");
+      expect(firstSearchResultTransaction.admin).toEqual(address);
+
+      // Delete domain
+      const deleteDomainTx = await connection.withDefaultFee<DeleteDomainTx>(
+        {
+          kind: "bns/delete_domain",
+          chainId: registryChainId,
+          domain,
```
Comment (webmaster128) [inline packages/iov-bns/src/bnsconnection.posttxs.spec.ts]: This shorthand notation is disallowed in IOV Core by tslint rule `"object-literal-shorthand": [true, "never"],`. So it must be `domain: domain,`

Motivation is that an inconsistent literal with shorthands and non-shorthands is harder to read.

**Pull request merged**